src/Gaffer/Animation.cpp

```diff
@@ -116,31 +123,58 @@ Animation::KeyPtr Animation::Key::setTime( float time )
 				assert( key->m_hook.is_linked() );
 				if( ! active )
 				{
+					assert( key->m_active == false );
```

Comment:
Seems like these asserts belong in the previous commit, since this commit is about the AnimationGadget.